### ISSUE TITLE
Make client WCAG-AA-compliant for contrast

### ISSUE THREAD

**Issue by lyzadanger** [hypothesis/client#1809]
Part of https://github.com/hypothesis/product-backlog/issues/1085

Some, but not all, contrast violations will be caught by our a11y linter and tests. Identify and remedy contrast violations in the client. This is part of the overall WCAG compliance a11y effort.

Known places to fix:

* [x] Button contrast when using grey background (https://github.com/hypothesis/client/pull/1808)
* [x] Quote (https://github.com/hypothesis/client/pull/1810)
* [x] "edited..." timestamp in annotation header (https://github.com/hypothesis/client/pull/1810)
* [ ] ~~`SearchStatusBar` search information/count~~ Actually, there is technically enough contrast here...
* [ ] ~~Disabled variant of `PublishControl` post button~~ We've decided not to rectify this disabled state
